exchanges/poloniex/poloniex.go

```diff
@@ -16,920 +16,1162 @@ import (
-		values.Set("immediateOrCancel", "1")
+func intervalToString(interval kline.Interval) (string, error) {
+	for x := range supportedIntervals {
+		if supportedIntervals[x].val == interval {
+			return supportedIntervals[x].key, nil
+		}
 	}
+	return "", kline.ErrUnsupportedInterval
+}
 
-	if fillOrKill {
-		values.Set("fillOrKill", "1")
+// WithdrawCurrency withdraws a currency to a specific delegated address
+func (e *Exchange) WithdrawCurrency(ctx context.Context, arg *WithdrawCurrencyRequest) (*Withdraw, error) {
+	if arg.Coin.IsEmpty() {
+		return nil, currency.ErrCurrencyCodeEmpty
 	}
-
-	return result, e.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, http.MethodPost, orderType, values, &result)
+	if arg.Amount <= 0 {
+		return nil, limits.ErrAmountBelowMin
+	}
+	if arg.Network == "" {
+		return nil, errInvalidWithdrawalChain
+	}
+	if arg.Address == "" {
+		return nil, errAddressRequired
+	}
+	var resp *Withdraw
+	return resp, e.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, authNonResourceIntensiveEPL, http.MethodPost, "/v2/wallets/withdraw", nil, arg, &resp)
 }
 
-// CancelExistingOrder cancels and order by orderID
-func (e *Exchange) CancelExistingOrder(ctx context.Context, orderID int64) error {
-	result := GenericResponse{}
-	values := url.Values{}
-	values.Set("orderNumber", strconv.FormatInt(orderID, 10))
-
-	err := e.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, http.MethodPost, poloniexOrderCancel, values, &result)
-	if err != nil {
-		return err
+// GetAccountMargin retrieves account margin information
+func (e *Exchange) GetAccountMargin(ctx context.Context, accountType string) (*AccountMargin, error) {
+	params := url.Values{}
+	if accountType != "" {
+		params.Set("accountType", accountType)
 	}
+	var resp *AccountMargin
+	return resp, e.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, authNonResourceIntensiveEPL, http.MethodGet, "/margin/accountMargin", params, nil, &resp)
+}
 
-	if result.Success != 1 {
-		return errors.New(result.Error)
+// GetBorrowStatus retrieves borrow status of currencies
+func (e *Exchange) GetBorrowStatus(ctx context.Context, ccy currency.Code) ([]*BorrowStatus, error) {
+	params := url.Values{}
+	if !ccy.IsEmpty() {
+		params.Set("currency", ccy.String())
 	}
-
-	return nil
+	var resp []*BorrowStatus
+	return resp, e.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, authNonResourceIntensiveEPL, http.MethodGet, "/margin/borrowStatus", params, nil, &resp)
 }
 
-// MoveOrder moves an order
-func (e *Exchange) MoveOrder(ctx context.Context, orderID int64, rate, amount float64, postOnly, immediateOrCancel bool) (MoveOrderResponse, error) {
-	result := MoveOrderResponse{}
-	values := url.Values{}
-
-	if orderID == 0 {
-		return result, errors.New("orderID cannot be zero")
+// MaximumBuySellAmount get maximum and available buy/sell amount for a given symbol.
+func (e *Exchange) MaximumBuySellAmount(ctx context.Context, symbol currency.Pair) (*MaxBuySellAmount, error) {
```

Comment:
🚧 Change request:
* Think this needs to mention margin
* Name doesn't convey that this can be used for available as well. Their endpoint naming doesn't, but that's their problem
* We have "get[space]" again here, and that's something we've tried to stamp out a few times in comments. Please search for "get " and fix all occurrences
* Considering if `GetMarginAmountAvailable` might work, too
* The type returned needs renaming
```suggestion
// GetMarginBuySellAmounts gets the maximum and available margin buy/sell amount for a given symbol
func (e *Exchange) GetMarginBuySellAmounts(ctx context.Context, symbol currency.Pair) (*MarginBuySellAmount, error) {
```